skcosmo/feature_selection/voronoi_fps.py

```diff
@@ -0,0 +1,163 @@
+            sparse_timing /= VORONOI_TRIALS     
+            if sparse_timing < full_timing:
+                a = self.full_fraction
+            else: 
+                b = self.full_fraction
+            print(self.full_fraction, sparse_timing, full_timing)        
+        self.full_fraction = a # make sure we are on the "good" side
+        self.stats = []
+        
+        super()._init_greedy_search(X, y, n_to_select)
+
+    def _continue_greedy_search(self, X, y, n_to_select):
+        """ Continues the search. Prepares an array to store the selected features. """
+
+        super()._continue_greedy_search(X, y, n_to_select)
+
+        self.sel_d2q_ = np.pad(self.sel_d2q_, (0, n_pad), "constant", constant_values=0)
+        self.new_dist_ = np.zeros(X.shape[1])
+
+    def _get_active(self, X, last_selected):
+        """
+        Finds the indices of the Voronoi cells that might change due to the fact
+        that a new point has been selected (index is last_selected)
+        """
+        # because we haven't called super() yet, self.n_selected_ has not been
+        # incremented yet
+
+        """must compute distance of the new point to all the previous FPS. Some
+            of these might have been computed already, but bookkeeping could be
+            worse that recomputing (TODO: verify!)
+            """
+        # calculation in a single block
+        self.sel_d2q_[: self.n_selected_] = (
+            self.norms_[self.selected_idx_[: self.n_selected_]]
+            + self.norms_[last_selected]
+            - 2 * (self.X_selected_[:, : self.n_selected_].T @ X[:, last_selected])
+        ) * 0.25
+
+        # these are the points for which need to recompute distances. 
+        # L is the last point selected
+        # S are the selected points from before this iteration
+        # X are the candidates
+        # the logic here is that we want to check if d(XL) can be smaller than 
+        # min(d(XS)) (which is stored in self.haussdorf_)
+        # now, if a point belongs to the Voronoi cell of S then min(d(XS_i))=d(XS).
+        # Triangle inequality implies that d(XL)>=|d(XS) - d(SL)| so we just need to
+        # check if |d(XS) - d(SL)|>= d(XS) to know that we don't need to check X.
+        # but |d(XS) - d(SL)|^2>= d(XS)^2 if and only if d(SL)/2 > d(SX)
+        active_points = np.where(
+                        self.sel_d2q_[self.vlocation_of_idx]
+                        <self.haussdorf_)[0]
+
+        return active_points
+
+    def _calculate_distances(self, X, last_selected, **kwargs):
+        
+        # n_selected has not been incremented, so index of new voronoi is
+        # n_selected
+
+        start = time()
+        if self.n_selected_ == 0:
+            self.haussdorf_ = super()._calculate_distances(X, last_selected)
+            # tracker of how many distances must be computed at each step                        
+            self.number_calculated_dist = np.shape(self.haussdorf_)[0]
```

Comment:
I knew it would turn up at some point :). It's actually possible to add it to the benchmarks and remove it from the main code, but it would require copying a lot of chunks of code.  I thought that it's the best solution